---
title: "Using GenesPackage"
author: "Arianna Rigamonti - 252321 - arianna1.rigamonti@mail.polimi.it"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Using GenesPackage}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

The GenesPackage provides a set of classes and methods to represent and manipulate different types of genes. This vignette provides an overview of the main functionalities of the package.

# Installation

To install the package, use the following commands:
```{r}
# Install the package from source
# install.packages("GenesPackage_0.1.0.tar.gz", repos = NULL, type = "source")

# Load the package
library(GenesPackage)
```

# Creating Gene Objects

The package provides various functions to create different types of gene objects. Below are some examples.

## Creating a Generic Gene
```{r}
library(GenomicRanges)
# Create a GRanges object for the gene structure
gr <- GRanges(seqnames = "chr1", ranges = IRanges(start = 1, end = 1000))

# Create a generic Gene object
gene <- createGene(ID = 1L, symbol = "SYMBOL", name = "Gene Name", 
                   description = "Description", chr = "chr1", start = 1, 
                   end = 1000, strand = "+", product = list())

# Display the gene object
gene
```

## Creating a lncRNA Gene
```{r}
# Create a lncRNAGene object
lncrna_gene <- createLncRNAGene(ID = 2L, symbol = "SYMBOL_LNC", 
                                name = "LncRNA Name", description = "LncRNA Description", 
                                chr = "chr1", start = 1, end = 1000, strand = "+", 
                                product = list(), lncRNAID = "lncrna1", 
                                RNASequence = "RNA_SEQUENCE")

# Display the lncRNA gene object
lncrna_gene
```

## Creating a microRNA Gene
```{r}
# Create a microRNAGene object
mirna_gene <- createMicroRNAGene(ID = 3L, symbol = "SYMBOL_MIR", 
                                 name = "MicroRNA Name", description = "MicroRNA Description", 
                                 chr = "chr1", start = 1, end = 1000, strand = "+", 
                                 product = list(), microRNAID = "mirna1", 
                                 microRNASequence = "SEED_SEQ")

# Display the microRNA gene object
mirna_gene
```

## Creating a piRNA Gene
```{r}
# Create a piRNAGene object
pirna_gene <- createPiRNAGene(ID = 8L, symbol = "SYMBOL_PI", 
                              name = "piRNA Name", description = "piRNA Description", 
                              chr = "chr1", start = 1, end = 1000, strand = "+", 
                              product = list(), piRNAID = "pirna1", 
                              piRNASequence = "PIRNA_SEQ")

# Display the piRNA gene object
pirna_gene
```

## Creating a Protein-Coding Gene
```{r}
# Create a ProteinCodingGene object
protein_gene <- createProteinCodingGene(ID = 1L, symbol = "SYMBOL", 
                                        name = "Gene Name", description = "Description", 
                                        chr = "chr1", start = 1, end = 1000, strand = "+", 
                                        product = list(), proteinID = "protein1", 
                                        proteinSequence = "SEQUENCE")

# Display the protein-coding gene object
protein_gene
```

## Creating an rRNA Gene
```{r}
# Create an rRNAGene object
rrna_gene <- createRRNAGene(ID = 7L, symbol = "SYMBOL_R", 
                            name = "rRNA Name", description = "rRNA Description", 
                            chr = "chr1", start = 1, end = 1000, strand = "+", 
                            product = list(), rRNAID = "rrna1", rRNASequence = "RRNA_SEQ")

# Display the rRNA gene object
rrna_gene
```

## Creating a siRNA Gene
```{r}
# Create a siRNAGene object
sirna_gene <- createSiRNAGene(ID = 4L, symbol = "SYMBOL_SI", 
                              name = "siRNA Name", description = "siRNA Description", 
                              chr = "chr1", start = 1, end = 1000, strand = "+", 
                              product = list(), siRNAID = "sirna1", 
                              siRNASequence = "SIRNA_SEQ")

# Display the siRNA gene object
sirna_gene
```

## Creating a snoRNA Gene
```{r}
# Create a snoRNAGene object
snorna_gene <- createSnoRNAGene(ID = 5L, symbol = "SYMBOL_SNO", 
                                name = "snoRNA Name", description = "snoRNA Description", 
                                chr = "chr1", start = 1, end = 1000, strand = "+", 
                                product = list(), snoRNAID = "snorna1", 
                                snoRNASequence = "SNORNA_SEQ")

# Display the snoRNA gene object
snorna_gene
```

## Creating a tRNA Gene
```{r}
# Create a tRNAGene object
trna_gene <- createTRNAGene(ID = 6L, symbol = "SYMBOL_T", 
                            name = "tRNA Name", description = "tRNA Description", 
                            chr = "chr1", start = 1, end = 1000, strand = "+", 
                            product = list(), tRNAID = "trna1", 
                            tRNASequence = "TRNA_SEQ")

# Display the tRNA gene object
trna_gene
```

# Accessing Gene Information

The package provides various methods to access information from gene objects. Below are some examples.

## Getting Gene ID
```{r}
# Get the gene ID
getID(gene)
```

## Getting Gene Symbol
```{r}
# Get the gene symbol
getSymbol(gene)
```

## Getting Gene Name
```{r}
# Get the gene name
getName(gene)
```

## Getting Gene Description
```{r}
# Get the gene description
getDescription(gene)
```

## Getting Gene Structure
```{r}
# Get the gene structure
getStructure(gene)
```

## Getting Gene Product
```{r}
# Get the gene product
getProduct(gene)
```

# Setting Gene Information

The package also provides methods to set information for gene objects.

## Setting Gene ID
```{r}
# Set a new gene ID
setID(gene) <- 2L
getID(gene)
```

## Setting Gene Symbol
```{r}
# Set a new gene symbol
setSymbol(gene) <- "NEW_SYMBOL"
getSymbol(gene)
```

## Setting Gene Name
```{r}
# Set a new gene name
setName(gene) <- "New Gene Name"
getName(gene)
```

## Setting Gene Description
```{r}
# Set a new gene description
setDescription(gene) <- "New Description"
getDescription(gene)
```

## Setting Gene Structure
```{r}
# Set a new gene structure
new_structure <- GRanges(seqnames = "chr2", ranges = IRanges(start = 100, end = 2000), strand = "-")
setStructure(gene) <- new_structure
getStructure(gene)
```

## Setting Gene Product
```{r}
# Set a new gene product
setProduct(gene) <- list("new product")
getProduct(gene)
```

# Summary of Gene Information

The package provides summary methods to get an overview of the gene information.

## Summary of a Gene
```{r}
# Summary of a generic Gene object
summary(gene)
```

## Summary of a lncRNA Gene
```{r}
# Summary of a lncRNA Gene object
summary(lncrna_gene)
```

## Summary of a microRNA Gene
```{r}
# Summary of a microRNA Gene object
summary(mirna_gene)
```

## Summary of a piRNA Gene
```{r}
# Summary of a piRNA Gene object
summary(pirna_gene)
```

## Summary of a Protein-Coding Gene
```{r}
# Summary of a Protein-Coding Gene object
summary(protein_gene)
```

## Summary of an rRNA Gene
```{r}
# Summary of an rRNA Gene object
summary(rrna_gene)
```

## Summary of a siRNA Gene
```{r}
# Summary of a siRNA Gene object
summary(sirna_gene)
```

## Summary of a snoRNA Gene
```{r}
# Summary of a snoRNA Gene object
summary(snorna_gene)
```

## Summary of a tRNA Gene
```{r}
# Summary of a tRNA Gene object
summary(trna_gene)
```

# Conclusion

The GenesPackage provides a comprehensive set of tools for representing and manipulating different types of genes in R. This vignette has provided an overview of the main functionalities of the package, including creating gene objects, accessing and setting gene information, and summarizing gene information.

For more detailed information about each function and class, please refer to the package documentation.

# Session Info

```{r}
sessionInfo()
```
